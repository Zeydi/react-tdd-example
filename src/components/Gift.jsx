import React, { Component } from "react";

class Gift extends Component {
  constructor() {
    super();
    this.state = { person: "", present: "" };
  }
  render() {
    return (
      <div>
        <form>
          <label>Person </label>
          <input
            className="input-person"
            onChange={(evt) => this.setState({ person: evt.target.value })}
          />
          <label>Present </label>
          <input
            className="input-present"
            onChange={(evt) => this.setState({ present: evt.target.value })}
          />
        </form>
        <button
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          remove gift
        </button>
      </div>
    );
  }
}

export default Gift;
