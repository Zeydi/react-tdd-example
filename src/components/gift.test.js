import React from "react";
import { shallow } from "enzyme";
import Gift from "./Gift";

describe("Gift component", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id: 1 }, removeGift: mockRemove };
  const gift = shallow(<Gift {...props} />);

  it("renders coreectly", () => {
    expect(gift).toMatchSnapshot();
  });

  it("initializes state with a person and present", () => {
    expect(gift.state()).toEqual({ person: "", present: "" });
  });

  describe("when typing into the person input", () => {
    const person = "sister";

    beforeEach(() => {
      gift
        .find(".input-person")
        .simulate("change", { target: { value: person } });
    });

    it("should updates state", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("when typing into the present input", () => {
    const present = "flower";

    beforeEach(() => {
      gift
        .find(".input-present")
        .simulate("change", { target: { value: present } });
    });

    it("should updates state", () => {
      expect(gift.state().present).toEqual(present);
    });
  });

  describe("when clicking the Remove gift button", () => {
    const present = "flower";

    beforeEach(() => {
      gift.find(".btn-remove").simulate("click");
    });

    it("calls the removegift callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});
