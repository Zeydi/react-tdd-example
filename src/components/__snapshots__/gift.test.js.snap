// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift component renders coreectly 1`] = `ShallowWrapper {}`;
